### PR TITLE
Add xml language server sample

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valeriy Svydenko <vsvydenk@redhat.com>
Provides a sample to demonstrate how to contribute new Language Server.
This sample registers **XML Language Server** and runs it automatically also it registers two commands to stop and to start XML LS.
The repository contains the jar which is produced  by the https://github.com/angelozerr/lsp4xml build.

### CODE REVIEW

File: xml-language-server-plugin/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "xml-language-server-plugin",
+  "publisher": "theia",
+  "keywords": [
+    "theia-plugin"
+  ],
+  "version": "0.0.1",
+  "license": "none",
```
Comment (benoitf) [inline xml-language-server-plugin/package.json]: in our case it should be EPL 2.0

File: xml-language-server-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    ],
+    "grammars": [
+      {
+        "scopeName": "text.xml",
+        "language": "xml",
+        "path": "./data/xml.tmLanguage.json"
+      }
+    ]
+  }
+}
```
Comment (benoitf) [inline xml-language-server-plugin/package.json]: FYI missing EOF (and on a lot of files)

File: xml-language-server-plugin/src/xml-language-server-backend.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+
+/**
```
Comment (benoitf) [inline xml-language-server-plugin/src/xml-language-server-backend.ts]: needs to add EPL 2.0 license for Che plug-ins

File: xml-language-server-plugin/data/xml.tmLanguage.json [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+        "captures": {
+          "0": {
+            "name": "punctuation.definition.comment.xml"
+          }
+        },
+        "end": "--%?>",
+        "name": "comment.block.xml"
+      }
+    }
+  }
```
Comment (mmorhun) [inline xml-language-server-plugin/data/xml.tmLanguage.json]: new line

File: xml-language-server-plugin/language-configuration.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            "open": "\"",
+            "close": "\""
+        },
+        {
+            "open": "'",
+            "close": "'"
+        }
+    ]
+}
```
Comment (mmorhun) [inline xml-language-server-plugin/language-configuration.json]: new line

File: xml-language-server-plugin/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        "experimentalDecorators": true,
+        "noUnusedLocals": true,
+        "emitDecoratorMetadata": true,
+        "downlevelIteration": true,
+        "module": "commonjs",
+        "moduleResolution": "node",
+        "target": "es5",
+        "lib": [
+            "es6",
+
```
Comment (mmorhun) [inline xml-language-server-plugin/tsconfig.json]: it is generator issue, but please remove unneeded lines and format this file

File: xml-language-server-plugin/src/xml-language-server-backend.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    const stopCommand = {
+        id: 'Stop XML LS',
+        label: "XML: Stop XML Language Server"
+    };
+
+    context.subscriptions.push(
+        theia.commands.registerCommand(startCommand, (...args: any[]) => {
+            context.subscriptions.push(theia.languageServer.registerLanguageServerProvider(lsInfo));
```
Comment (mmorhun) [inline xml-language-server-plugin/src/xml-language-server-backend.ts]: What if it invoked twice or more? The same about stop.

Comment (svor) [reply xml-language-server-plugin/src/xml-language-server-backend.ts]: Can you clarify please what do you mean?

**Comment by svor** [comment]
@benoitf I've added a script to download the jar.

File: xml-language-server-plugin/src/xml-language-server-backend.ts (hunk 7 above)
Comment (benoitf) [reply xml-language-server-plugin/src/xml-language-server-backend.ts]: I would say: if user invoke two times "start LS" command or call "stop" before "start"

Comment (svor) [reply xml-language-server-plugin/src/xml-language-server-backend.ts]: If user invokes two times start,  the implementation of API will take care of it. There will be message like 'ls with this id is already registered'. 
If user calls stop before start, the API will check that ls with this ID isn't registered and nothing will happen.

Comment (mmorhun) [reply xml-language-server-plugin/src/xml-language-server-backend.ts]: Thanks for explanation

**Pull request merged**